frontend/packages/dev-console/src/components/edit-deployment/EditDeploymentForm.tsx

```diff
@@ -0,0 +1,137 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { FormikProps, FormikValues } from 'formik';
+import { safeLoad } from 'js-yaml';
+import { saveAs } from 'file-saver';
+import {
+  FlexForm,
+  FormBody,
+  FormFooter,
+  FormHeader,
+  SyncedEditorField,
+  YAMLEditorField,
+} from '@console/shared/src';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import EditDeploymentFormEditor from './EditDeploymentFormEditor';
+import { getResourcesType } from '../edit-application/edit-application-utils';
+import { EditorType } from '@console/shared/src/components/synced-editor/editor-toggle';
+import * as _ from 'lodash';
+import { Resources } from '../import/import-types';
+import { DeploymentConfigModel, DeploymentModel } from '@console/internal/models';
+import { history } from '@console/internal/components/utils';
+import {
+  convertDeploymentToEditForm,
+  convertEditFormToDeployment,
+} from './utils/edit-deployment-utils';
+import { safeJSToYAML } from '@console/shared/src/utils/yaml';
+
+const EditDeploymentForm: React.FC<FormikProps<FormikValues> & {
+  heading: string;
+  resource: K8sResourceKind;
```

Comment:
This is used for both `Deployment` and `DeploymentConfig` and hence `DeploymentKind` cannot be used here as the two have a few different attributes like DeploymentStrategy, Triggers etc.

Reply:
Okies, I guess we can continue with `K8sResourceKind`... it's probably something we should really get away from relying on because it helps us in almost no way other than `has metadata`.